src/lib.rs

```diff
@@ -77,21 +80,35 @@ where
         self.read_register(Register::WHOAMI)
     }
 
+    /// X,Y,Z-axis enable.
+    /// `CTRL_REG1`: `Xen`, `Yen`, `Zen`
+    pub fn enable_axis(&mut self, (x, y, z): (bool, bool, bool)) -> Result<(), Error<E>> {
+        self.modify_register(Register::CTRL1, |mut ctrl1| {
+            ctrl1 &= !(X_EN | Y_EN | Z_EN); // disable all axes
+
+            ctrl1 |= if x { X_EN } else { 0 };
+            ctrl1 |= if y { Y_EN } else { 0 };
+            ctrl1 |= if z { Z_EN } else { 0 };
+
+            ctrl1
+        })
+    }
+
     /// Operating mode selection.
     /// `CTRL_REG1`: `LPen` bit, `CTRL_REG4`: `HR` bit
     pub fn set_mode(&mut self, mode: Mode) -> Result<(), Error<E>> {
         match mode {
             Mode::LowPower => {
-                self.register_set_bits(Register::CTRL1, 0x8)?;
-                self.register_reset_bits(Register::CTRL4, 0x8)?;
+                self.register_set_bits(Register::CTRL1, LP_EN)?;
+                self.register_clear_bits(Register::CTRL4, HR)?;
             }
             Mode::Normal => {
-                self.register_reset_bits(Register::CTRL1, 0x8)?;
-                self.register_reset_bits(Register::CTRL4, 0x8)?;
+                self.register_clear_bits(Register::CTRL1, LP_EN)?;
+                self.register_clear_bits(Register::CTRL4, HR)?;
             }
             Mode::HighResolution => {
-                self.register_reset_bits(Register::CTRL1, 0x8)?;
-                self.register_set_bits(Register::CTRL4, 0x8)?;
+                self.register_clear_bits(Register::CTRL1, LP_EN)?;
+                self.register_set_bits(Register::CTRL4, HR)?;
             }
         }
 
```

Comment:
3.2.1 table 11 says theres a turn on time to switching into modes
Might want to busy wait or 
12-bit mode to 8-bit mode1/ODR
12-bit mode to 10-bit mode1/ODR
10-bit mode to 8-bit mode1/ODR
10-bit mode to 12-bit mode7/ODR
8-bit mode to 10-bit mode1/ODR
8-bit mode to 12-bit mode7/ODR
(ODR is the data rate)